Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp

```diff
@@ -67,6 +68,216 @@ ReadonlySpan<StringView> available_calendars()
     return calendars.span();
 }
 
+// 12.2.2 CreateCalendarMethodsRecord ( calendar, methods ), https://tc39.es/proposal-temporal/#sec-temporal-createcalendarmethodsrecord
+ThrowCompletionOr<CalendarMethodsRecord> create_calendar_methods_record(VM& vm, Variant<String, NonnullGCPtr<Object>> const& calendar, Vector<CalendarMethod> const& methods)
+{
+    // 1. Let record be the Calendar Methods Record { [[Receiver]]: calendar, [[DateAdd]]: undefined, [[DateFromFields]]: undefined, [[DateUntil]]: undefined, [[Day]]: undefined, [[Fields]]: undefined, [[MergeFields]]: undefined, [[MonthDayFromFields]]: undefined, [[YearMonthFromFields]]: undefined }.
+    CalendarMethodsRecord record {
+        .receiver = calendar,
+        .date_add = nullptr,
+        .date_from_fields = nullptr,
+        .date_until = nullptr,
+        .day = nullptr,
+        .fields = nullptr,
+        .merge_fields = nullptr,
+        .month_day_from_fields = nullptr,
+        .year_month_from_fields = nullptr,
+    };
+
+    // 2. For each element methodName in methods, do
+    for (auto const& method_name : methods) {
+        // a. Perform ? CalendarMethodsRecordLookup(record, methodName).
+        TRY(calendar_methods_record_lookup(vm, record, method_name));
+    }
+
+    // 3. Return record.
+    return record;
+}
+
+ThrowCompletionOr<Optional<CalendarMethodsRecord>> create_calendar_methods_record_from_relative_to(VM& vm, PlainDate* plain_relative_to, ZonedDateTime* zoned_relative_to, Vector<CalendarMethod> const& methods)
+{
+    // FIXME: The casts to NonnullGCPtr<Object> should not be here, and can be fixed once PlainDate & ZonedDateTime have the updated type in the [[Calendar]] slot.
+
+    // 1. If zonedRelativeTo is not undefined, return ? CreateCalendarMethodsRecord(zonedRelativeTo.[[Calendar]], methods).
+    if (zoned_relative_to)
+        return TRY(create_calendar_methods_record(vm, NonnullGCPtr<Object> { zoned_relative_to->calendar() }, methods));
+
+    // 2. If plainRelativeTo is not undefined, return ? CreateCalendarMethodsRecord(plainRelativeTo.[[Calendar]], methods).
+    if (plain_relative_to)
+        return TRY(create_calendar_methods_record(vm, NonnullGCPtr<Object> { plain_relative_to->calendar() }, methods));
+
+    // 3. Return undefined.
+    return OptionalNone {};
+}
+
+// 12.2.4 CalendarMethodsRecordLookup ( calendarRec, methodName ), https://tc39.es/proposal-temporal/#sec-temporal-calendarmethodsrecordlookup
+ThrowCompletionOr<void> calendar_methods_record_lookup(VM& vm, CalendarMethodsRecord& calendar_record, CalendarMethod method_name)
+{
+    auto& realm = *vm.current_realm();
+
+    // 1. Assert: CalendarMethodsRecordHasLookedUp(calendarRec, methodName) is false.
```

Comment:
Missing: 
```c++
VERIFY(calendar_methods_record_has_looked_up(calendar_record, method_name));
```

Reply:
This is implemented by the `VERIFY` below - which I did that way since it's a little more efficient, and also give a better assertion failure if it fails

Reply:
Gotcha, missed that.